File: Frontend/SitioSeries2/src/app/core/models/serie.ts
import { Category } from './category';

export interface Serie{
    id?: number;
    titulo: string;
    descripcion:string;
    puntuacion: number;
    imgPortada: string;
    categoriaId: number;
    nombreCategoria:string;
    anio:number;
    duracion:number;
    temporada:number;
    capitulo:number;
    paisId:number;
    link:string;
    usuarioId:number;
}

File: Backend/Api/Entities/Models/Serie.cs
namespace Entities.Models
{
    public partial class Serie
    {
        public int Id { get; set; }
        public string Titulo { get; set; } = null!;
        public string Descripcion { get; set; } = null!;
        public float Puntuacion { get; set; }
        public string ImgPortada { get; set; } = null!;
        public int Anio { get; set; }
        public int Duracion { get; set; }
        public int Temporada { get; set; }
        public int Capitulo { get; set; }
        public string Link { get; set; } = null!;

        public int CategoriaId { get; set; }
        public int PaisId { get; set; }
        public int UsuarioId { get; set; }

        public  Categoria Categoria { get; set; } = null!;
        public Pais Pais { get; set; } = null!;
        public Usuario? Usuario { get; set; }

        public ICollection<Comentario> Comentarios { get; set; }
        public ICollection<SeriesFavoritas> SeriesFavoritas { get; set; }
    }
}
